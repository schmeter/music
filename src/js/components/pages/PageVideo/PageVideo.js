import React from 'react';
import PropTypes from 'prop-types';

import Page from '../../Page';
import Video from './Video';

const PageVideo = ({ videos }) => (
    <Page id="video">
        <div className="videos">
            {videos.map(video => !video.hidden && (
                <Video
                    key={video.id}
                    video={video}
                />
            ))}
        </div>
    </Page>
);

PageVideo.propTypes = {
    videos: PropTypes.array.isRequired,
};

export default PageVideo;
