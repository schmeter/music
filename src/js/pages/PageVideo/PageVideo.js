import React from 'react';
import PropTypes from 'prop-types';

import Page from '../../components/Page';
import Video from '../../components/Video';

const PageVideo = ({ videos }) => (
    <Page id="video">
        {videos.map(video => !video.hidden && (
            <Video
                key={video.id}
                video={video}
            />
        ))}
    </Page>
);

PageVideo.propTypes = {
    videos: PropTypes.array.isRequired,
};

export default PageVideo;
