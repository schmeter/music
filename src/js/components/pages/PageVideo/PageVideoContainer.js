import { connect } from 'react-redux';

import PageVideo from './PageVideo';
import { getVideos } from '../../../state/selectors/video';

const mapStateToProps = state => ({
    videos: getVideos(state),
});

export default connect(mapStateToProps, null)(PageVideo);
