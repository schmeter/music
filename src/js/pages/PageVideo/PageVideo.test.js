import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageVideo from './PageVideo';

jest.mock('../../components/Page', () => () => <div />);
jest.mock('../../components/Video', () => () => <div />);

describe('PageVideo', () => {
    it('renders correctly', () => {
        const props = {
            videos: [{
                id: 'test',
            }, {
                hidden: true,
                id: 'test2',
            }],
        };
        const component = shallow(<PageVideo {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
