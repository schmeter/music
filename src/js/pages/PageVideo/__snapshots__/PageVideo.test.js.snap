// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageVideo renders correctly 1`] = `
<Component
  id="video"
>
  <Component
    key="test"
    video={
      Object {
        "id": "test",
      }
    }
  />
</Component>
`;
